proto/workflow_execution.proto

```diff
@@ -8,58 +8,118 @@ import "google/api/annotations.proto";
 import "google/protobuf/struct.proto";
 import "task_execution.proto";
 
+//available workflow types supported by this WES
+enum WorkflowType {
+    UNKNOWN = 0;
+    CWL = 1;
+    WDL = 2;
+}
+
+//Return envelope for workflow listing
+message WorkflowListResponse {
+   repeated JobDesc workflows = 1;
+   string nextPageToken = 2;
+}
+
+//Small description of workflows, returned by server during listing
+message JobDesc {
+  //REQUIRED
+  string workflow_id  = 1;
+  //REQUIRED
+  ga4gh_task_exec.State state = 2;
+}
 
+//workflow request object
 message WorkflowRequest {
-  string workflow_url = 1;
-  google.protobuf.Struct input = 2;
+  //OPTIONAL
+  //user name of the workflow
+  string name = 1;
+  //OPTIONAL
+  //parameter for execution engine to define/store group information
+  string project_id = 2;
```

Comment:
OK, I removed name and project_Id (and description) since all of these can be modeled as key-values.